drivers/gpio/gpio_kscan_ite_it8xxx2.c

```diff
@@ -0,0 +1,244 @@
+
+		/* pullup or no pull */
+		if (*reg_ksi_kso_gpod & mask) {
+			flags |= GPIO_PULL_UP;
+		}
+	}
+
+	*out_flags = flags;
+
+	return 0;
+}
+#endif
+
+static int gpio_kscan_it8xxx2_port_get_raw(const struct device *dev,
+					   gpio_port_value_t *value)
+{
+	const struct gpio_kscan_cfg *const config = dev->config;
+	volatile uint8_t *reg_ksi_kso_gdmr = config->reg_ksi_kso_gdmr;
+
+	/* Get physical level from all pins of the port */
+	*value = *reg_ksi_kso_gdmr;
+
+	return 0;
+}
+
+static int gpio_kscan_it8xxx2_port_set_masked_raw(const struct device *dev,
+						  gpio_port_pins_t mask,
+						  gpio_port_value_t value)
+{
+	const struct gpio_kscan_cfg *const config = dev->config;
+	volatile uint8_t *reg_ksi_kso_gdat = config->reg_ksi_kso_gdat;
+	uint8_t out = *reg_ksi_kso_gdat;
+
+	/* Set high/low level to mask pins of the port */
+	*reg_ksi_kso_gdat = ((out & ~mask) | (value & mask));
+
+	return 0;
+}
+
+static int gpio_kscan_it8xxx2_port_set_bits_raw(const struct device *dev,
+						gpio_port_pins_t pins)
+{
+	const struct gpio_kscan_cfg *const config = dev->config;
+	volatile uint8_t *reg_ksi_kso_gdat = config->reg_ksi_kso_gdat;
+
+	/* Set high level to pins of the port */
+	*reg_ksi_kso_gdat |= pins;
+
+	return 0;
+}
+
+static int gpio_kscan_it8xxx2_port_clear_bits_raw(const struct device *dev,
+						  gpio_port_pins_t pins)
+{
+	const struct gpio_kscan_cfg *const config = dev->config;
+	volatile uint8_t *reg_ksi_kso_gdat = config->reg_ksi_kso_gdat;
+
+	/* Set low level to pins of the port */
+	*reg_ksi_kso_gdat &= ~pins;
+
+	return 0;
+}
+
+static int gpio_kscan_it8xxx2_port_toggle_bits(const struct device *dev,
+					       gpio_port_pins_t pins)
+{
+	const struct gpio_kscan_cfg *const config = dev->config;
+	volatile uint8_t *reg_ksi_kso_gdat = config->reg_ksi_kso_gdat;
+
+	/* Toggle output level to pins of the port */
+	*reg_ksi_kso_gdat ^= pins;
+
+	return 0;
+}
+
+static int gpio_kscan_it8xxx2_init(const struct device *dev)
+{
+	return 0;
+}
+
+static const struct gpio_driver_api gpio_kscan_it8xxx2_driver_api = {
+	.pin_configure = gpio_kscan_it8xxx2_configure,
+#ifdef CONFIG_GPIO_GET_CONFIG
+	.pin_get_config = gpio_kscan_it8xxx2_get_config,
+#endif
+	.port_get_raw = gpio_kscan_it8xxx2_port_get_raw,
+	.port_set_masked_raw = gpio_kscan_it8xxx2_port_set_masked_raw,
+	.port_set_bits_raw = gpio_kscan_it8xxx2_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_kscan_it8xxx2_port_clear_bits_raw,
+	.port_toggle_bits = gpio_kscan_it8xxx2_port_toggle_bits,
+};
+
+#define GPIO_KSCAN_IT8XXX2_INIT(inst)                                     \
+static const struct gpio_kscan_cfg gpio_kscan_it8xxx2_cfg_##inst = {      \
+	.common = {                                                       \
+		.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_NGPIOS(          \
+				 DT_INST_PROP(inst, ngpios))              \
+	},                                                                \
+	.reg_ksi_kso_goen = (uint8_t *)DT_INST_REG_ADDR_BY_IDX(inst, 0),  \
```

Comment:
You can use `DT_INST_REG_ADDR_BY_NAME()` here instead.
```suggestion
	.reg_ksi_kso_goen = (uint8_t *)DT_INST_REG_ADDR_BY_NAME(goen, inst),  \
```

Reply:
Done